feat: Orchestration with Prompt Registry (#514)
SAP/ai-sdk-js: packages/orchestration/README.md

## Usage

Leverage the orchestration service capabilities by using the orchestration client.
You can perform a simple prompt by running:

```ts
import { OrchestrationClient } from '@sap-ai-sdk/orchestration';

const orchestrationClient = new OrchestrationClient({
  llm: {
    model_name: 'gpt-4o'
  },
  templating: {
    template: [
      { role: 'user', content: 'Hello World! Why is this phrase so famous?' }
    ]
  }
});

const response = await orchestrationClient.chatCompletion();
console.log(response.getContent());
```

Here we use GPT-4o with a single user message as prompt and print out the response.

> The `OrchestrationClient` allows you to combine various modules, such as templating and content filtering, while sending chat completion requests to an orchestration-compatible generative AI model.

Reviewer: ```suggestion
The `OrchestrationClient` enables combining various modules, such as templating and content filtering, while sending chat completion requests to an orchestration-compatible generative AI model.
```

In addition to the examples below, you can find more **sample code** [here](https://github.com/SAP/ai-sdk-js/blob/main/sample-code/src/orchestration.ts).

> ### LLM Config

Reviewer: [req] Please update the TOC

> Configure which LLM to use module by setting the `model_name` property.

Reviewer: ```suggestion
Choose the LLM by setting the `model_name` property.
```

> Define the optional `model_version` property to choose an available model version.

Reviewer: ```suggestion
```

> By default, the version is set to `latest`.

Reviewer: ```suggestion
```

> Specify the optional `model_params` property to apply specific parameters to the model

Reviewer: ```suggestion
Optionally, define `model_version` (default: `latest`) and `model_params` for custom settings.
```

[pp] Shortening the sentences, as they sometimes feel toooooo verbose.

> [!Tip]
>
> #### Harmonized API
>

> > As the orchestration service API is harmonized, you can switch to a different model, even from another vendor, by changing only the `model_name` property.

Reviewer: ```suggestion
> Since the Orchestration Service API is harmonized, switching to a different model—even from another vendor—only requires changing the `model_name` property.
```

> > Here’s an example where only one line of code is changed.

Reviewer: [pp] I feel like we dont need this. The above sentence say only one change is needed and we highlight that in the code

Author: Agreed. After reviewing this I felt that we can rather just link to the docs of orchestration where this is explained better anyway, so I shortened this to just a short explanation + link

);
```

### Prompt Registry

> Alternatively to passing a template in the request, one can also use prepared templates stored in the [Prompt Registry](https://help.sap.com/docs/sap-ai-core/sap-ai-core-service-guide/prompt-registry) of SAP AI Core.

Reviewer: ```suggestion
Alternatively, prepared templates from the [Prompt Registry](https://help.sap.com/docs/sap-ai-core/sap-ai-core-service-guide/prompt-registry) of SAP AI Core can be used instead of passing a template in the request.
```

```ts
const orchestrationClient = new OrchestrationClient({
  llm,
  templating: {
    template_ref: {
      name: 'get-capital',
      scenario: 'e2e-test',
      version: '0.0.1'
    }
  }
});

return orchestrationClient.chatCompletion({
  inputParams: { input: 'France' }
});
```

> You can reference a prompt template either by ID or name, scenario and version.

Reviewer: ```suggestion
A prompt template can be referenced either by ID, or by using a combination of name, scenario, and version.
```

[pp] Just to make it more explicit.

> For more information on how to store a template in the Prompt Registry, please refer to [this guide](https://help.sap.com/docs/sap-ai-core/sap-ai-core-service-guide/create-prompt-template-imperative).

Reviewer: ```suggestion
For details on storing a template in the Prompt Registry, refer to [this guide](https://help.sap.com/docs/sap-ai-core/sap-ai-core-service-guide/create-prompt-template-imperative).
```

[style-guide] avoid please
